flutter/lib/src/integrations/load_image_list_integration.dart

```diff
@@ -33,12 +34,44 @@ class _LoadImageListIntegrationEventProcessor implements EventProcessor {
   @override
   Future<SentryEvent?> apply(SentryEvent event, Hint hint) async {
     if (event.needsSymbolication()) {
-      final images = await _native.loadDebugImages();
+      Set<String> imageAddresses = {};
+      var exceptions = event.exceptions;
+      if (exceptions != null && exceptions.isNotEmpty) {
+        for (var e in exceptions) {
+          if (e.stackTrace != null) {
+            imageAddresses.addAll(
+              _collectImageAddressesFromStackTrace(e.stackTrace!),
+            );
+          }
+        }
+      }
+
+      if (event.threads != null && event.threads!.isNotEmpty) {
+        for (var thread in event.threads!) {
+          if (thread.stacktrace != null) {
+            imageAddresses.addAll(
+              _collectImageAddressesFromStackTrace(thread.stacktrace!),
+            );
+          }
+        }
+      }
+
+      final images = await _native.loadDebugImages(imageAddresses);
       if (images != null) {
         return event.copyWith(debugMeta: DebugMeta(images: images));
       }
     }
 
     return event;
   }
+
+  Set<String> _collectImageAddressesFromStackTrace(SentryStackTrace trace) {
+    Set<String> imageAddresses = {};
+    for (var frame in trace.frames) {
+      if (frame.imageAddr != null) {
+        imageAddresses.add(frame.imageAddr!);
```

Comment:
Actually, this isn't going to work because for obfuscated code we may only have instruction addresses, not image addresses.

Reply:
Thanks for the hint, I wasn`t aware of that.

Reply:
we need to use the instruction addresses to grab the images like so

https://github.com/getsentry/sentry-react-native/blob/59a53cbd0983e4bc111169feedd7c8d7d19df9b6/ios/RNSentry.mm#L220C1-L225C1